File: go/415.go
func addStrings(num1 string, num2 string) string {
    var ret []byte
    var carry byte
    i, j := len(num1)-1, len(num2)-1
    for i >= 0 && j >= 0 {
        temp := num1[i] - '0' + num2[j] - '0' + carry
        ret = append(ret, temp%10+'0')
        carry = temp / 10
        i--
        j--
    }
    for i >= 0 {
        temp := num1[i] - '0' + carry
        ret = append(ret, temp%10+'0')
        carry = temp / 10
        i--
    }
    for j >= 0 {
        temp := num2[j] - '0' + carry
        ret = append(ret, temp%10+'0')
        carry = temp / 10
        j--
    }
    if carry != 0 {
        ret = append(ret, '1')
    }
    size := len(ret)
    for i := 0; i < size/2; i++ {
        ret[i], ret[size-i-1] = ret[size-i-1], ret[i]
    }
    return string(ret)
}

File: cpp/415.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int s1 = num1.size() - 1, s2 = num2.size() - 1, carry = 0;
        string ret;
        for (; s1 >= 0 && s2 >= 0; --s1,--s2) {
            int temp = num1[s1] - '0' + num2[s2] - '0' + carry;
            ret.push_back(temp%10+'0');
            carry = temp / 10;
        }
        for (; s1 >= 0; --s1) {
            int temp = num1[s1] - '0' + carry;
            ret.push_back(temp%10+'0');
            carry = temp / 10;
        }
        for (; s2 >= 0; --s2) {
            int temp = num2[s2] - '0' + carry;
            ret.push_back(temp%10+'0');
            carry = temp / 10;
        }
        if (carry)
            ret.push_back('1');
        int size = ret.size();
        for (int i = 0; i < size / 2; ++i)
            swap(ret[i], ret[size-i-1]);
        return ret;
    }

};
